src/plugins/saved_objects/public/finder/saved_object_finder.tsx

```diff
@@ -178,6 +191,25 @@ class SavedObjectFinderUi extends React.Component<
       })
     );
 
+    // Filter search objects whose language are not supported by the current Application
+    savedObjects = savedObjects.filter((obj) => {
```

Comment:
i think we should filter the results similar to how it does `(obj.type === 'index-pattern')` on line 180. that way we keep const savedObjects

Reply:
Consolidated the logic in one location.